modules/sdk/gradle-plugins-jsdoc/src/main/java/com/liferay/gradle/plugins/jsdoc/AppJSDocPlugin.java

```diff
@@ -0,0 +1,185 @@
+					Set<Project> subprojects =
+						appJSDocConfigurationExtension.getSubprojects();
+
+					if (subprojects.contains(subproject) &&
+						_hasJSDocSource(subproject)) {
+
+						_configureTaskAppJSDoc(appJSDocTask, subproject);
+					}
+				}
+
+			});
+	}
+
+	private JSDocTask _addTaskAppJSDoc(Project project) {
+		final JSDocTask jsDocTask = GradleUtil.addTask(
+			project, APP_JSDOC_TASK_NAME, JSDocTask.class);
+
+		jsDocTask.setDescription(
+			"Generates the API documentation for the JavaScript code in this " +
+				"app.");
+
+		jsDocTask.setDestinationDir(
+			new Callable<File>() {
+
+				@Override
+				public File call() throws Exception {
+					Project project = jsDocTask.getProject();
+
+					return new File(project.getBuildDir(), "jsdoc");
+				}
+
+			});
+
+		jsDocTask.setGroup(JavaBasePlugin.DOCUMENTATION_GROUP);
+
+		return jsDocTask;
+	}
+
+	private DownloadNodeModuleTask _addTaskDownloadAppJSDoc(Project project) {
+		DownloadNodeModuleTask downloadNodeModuleTask = GradleUtil.addTask(
+			project, DOWNLOAD_APP_JSDOC_TASK_NAME,
+			DownloadNodeModuleTask.class);
+
+		downloadNodeModuleTask.args("--no-save");
+		downloadNodeModuleTask.setDescription("Downloads JSDoc.");
+		downloadNodeModuleTask.setModuleName("jsdoc");
+		downloadNodeModuleTask.setModuleVersion(_VERSION);
+
+		return downloadNodeModuleTask;
+	}
+
+	private Jar _addTaskJarAppJSDoc(JSDocTask jsDocTask) {
+		Jar jar = GradleUtil.addTask(
+			jsDocTask.getProject(), JAR_APP_JSDOC_TASK_NAME, Jar.class);
+
+		jar.from(jsDocTask);
+		jar.setClassifier("jsdoc");
+		jar.setDescription(
+			"Assembles a jar archive containing the JavaScript documentation " +
+				"files for this app.");
+		jar.setGroup(BasePlugin.BUILD_GROUP);
+
+		return jar;
+	}
+
+	private void _configureTaskAppJSDoc(
+		JSDocTask appJSDocTask, DownloadNodeModuleTask downloadAppJSDocTask) {
+
+		appJSDocTask.dependsOn(downloadAppJSDocTask);
+	}
+
+	private void _configureTaskAppJSDoc(
+		JSDocTask appJSDocTask, Project subproject) {
+
+		JSDocTask subprojectJSDocTask = (JSDocTask)GradleUtil.getTask(
+			subproject, JSDocPlugin.JSDOC_TASK_NAME);
+
+		appJSDocTask.dependsOn(subprojectJSDocTask);
+
+		appJSDocTask.sourcesDirs(subprojectJSDocTask.getScriptFile());
+		appJSDocTask.sourcesDirs(subprojectJSDocTask.getSourcesDirs());
+	}
+
+	private boolean _hasJSDocSource(Project subproject) {
+		PluginContainer pluginContainer = subproject.getPlugins();
+
+		if (!pluginContainer.hasPlugin(JSDocPlugin.class)) {
+			return false;
+		}
+
+		SourceSet sourceSet = GradleUtil.getSourceSet(
+			subproject, SourceSet.MAIN_SOURCE_SET_NAME);
+
+		SourceDirectorySet sourceDirectorySet = sourceSet.getResources();
+
+		Set<File> srcDirs = sourceDirectorySet.getSrcDirs();
+
+		Iterator<File> iterator = srcDirs.iterator();
+
+		File file = new File(iterator.next(), "META-INF/resources");
+
+		if (file.exists()) {
+			return true;
+		}
+
+		return false;
+	}
+
+	private static final String _VERSION = "3.5.5";
```

Comment:
Instead of repeating the version here, maybe we can create a `com.liferay.gradle.plugins.jsdoc.internal.constants.JSDocConstants` class with a `VERSION` constant.